opentracing-api-extensions-tracer/src/main/java/io/opentracing/contrib/api/tracer/APIExtensionsSpanBuilder.java

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright 2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.contrib.api.tracer;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+
+import io.opentracing.ActiveSpan;
+import io.opentracing.BaseSpan;
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.Tracer.SpanBuilder;
+import io.opentracing.contrib.api.TracerObserver;
+
+public class APIExtensionsSpanBuilder implements SpanBuilder {
+
+    private final Tracer tracer;
+    private final List<TracerObserver> observers;
+
+    private final String operationName;
+    private final SpanBuilder wrappedBuilder;
+    private long startTimestampMicro = TimeUnit.MILLISECONDS.toMicros(System.currentTimeMillis());
+    private long startTimeNano = System.nanoTime();
+    private final Map<String,Object> tags = new ConcurrentHashMap<String,Object>();
+
+    public APIExtensionsSpanBuilder(Tracer tracer, List<TracerObserver> observers,
+            String operationName, SpanBuilder builder) {
+        this.tracer = tracer;
+        this.observers = observers;
```

Comment:
I suppose that still leaves the potential for the app to add/remove an observer which would impact inflight spans. So probably better to clone to make sure the observer sees the full lifecycle of the span being observed.